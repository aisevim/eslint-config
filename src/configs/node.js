import nodePlugin from 'eslint-plugin-n'

import { JS_GLOB } from '../constants.js'

export async function node() {
  return {
    files: [JS_GLOB],
    plugins: {
      node: nodePlugin,
    },
    rules: {
      'node/handle-callback-err': ['error', '^(err|error)$'],
      'node/no-deprecated-api': 'error',
      'node/no-exports-assign': 'error',
      'node/no-new-require': 'error',
      'node/no-path-concat': 'error',
      'node/prefer-promises/fs': 'error',
      'node/process-exit-as-throw': 'error',
    },
  }
}
