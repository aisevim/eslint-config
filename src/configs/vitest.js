import { GLOBS_TESTS } from '../globs.js'
import { createConfig, interopDefault } from '../utils.js'

export async function vitestConfig({ options = {} }) {
  const [pluginVitest, pluginNoOnlyTests] = await Promise.all([
    interopDefault(import('eslint-plugin-vitest')),
    interopDefault(import('eslint-plugin-no-only-tests')),
  ])

  return createConfig(options, {
    files: [GLOBS_TESTS],
    plugins: {
      'vitest': pluginVitest,
      'no-only-tests': pluginNoOnlyTests,
    },
    languageOptions: {
      globals: {
        ...pluginVitest.environments.env.globals,
      },
    },
    rules: {
      ...pluginVitest.configs.recommended.rules,
      'vitest/consistent-test-it': 'error',
      'vitest/no-done-callback': 'error',
      'vitest/no-duplicate-hooks': 'error',
      'vitest/no-standalone-expect': {
        additionalTestBlockFunctions: ['test'],
      },
      'vitest/no-test-return-statement': 'error',
      'vitest/prefer-comparison-matcher': 'error',
      'vitest/prefer-expect-resolves': 'error',
      'vitest/prefer-hooks-in-order': 'error',
      'vitest/prefer-hooks-on-top': 'error',
      'vitest/prefer-to-be-falsy': 'error',
      'vitest/prefer-to-be-object': 'error',
      'vitest/prefer-to-be-truthy': 'error',
      'vitest/prefer-to-contain': 'error',
      'vitest/prefer-to-have-length': 'error',
      'vitest/valid-describe-callback': 'error',

      'no-only-tests/no-only-tests': 'error',
    },
  })
}
