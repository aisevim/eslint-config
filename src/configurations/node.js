import nodePlugin from 'eslint-plugin-n'

export async function node() {
	return {
		files: ["**/*.?([cm])[jt]s?(x)"],
		plugins: {
			node: nodePlugin
		},
		rules: {
			'node/handle-callback-err': ['error', '^(err|error)$'],
			'node/no-deprecated-api': 'error',
			'node/no-exports-assign': 'error',
			'node/no-new-require': 'error',
			'node/no-path-concat': 'error',
			'node/prefer-promises/fs': 'error',
			'node/process-exit-as-throw': 'error',
		}
	}
}
