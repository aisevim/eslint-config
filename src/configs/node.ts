import node from 'eslint-plugin-n'

export async function node() {
	return {
		files: ["**/*.?([cm])[jt]s?(x)"],
		plugins: {
			n: node
		},
		rules: {
			'n/handle-callback-err': ['error', '^(err|error)$'],
			'n/no-deprecated-api': 'error',
			'n/no-exports-assign': 'error',
			'n/no-new-require': 'error',
			'n/no-path-concat': 'error',
			'n/prefer-promises/fs': 'error',
			'n/process-exit-as-throw': 'error',
		}
	}
}
