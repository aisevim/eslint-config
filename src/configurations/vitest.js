import vitest from "eslint-plugin-vitest";

export default [
  {
    files: ["tests/**", "**/__tests__/**"],
    plugins: {
      vitest,
    },
		languageOptions: {
			globals: {
				...vitest.environments.env.globals,
			},
		},
    rules: {
      ...vitest.configs.recommended.rules,
			'vitest/consistent-test-it': 'error',
			'vitest/no-done-callback': 'error',
			'vitest/no-duplicate-hooks': 'error',
			"vitest/no-standalone-expect": {
				"additionalTestBlockFunctions": ["test"]
			},
			'vitest/no-test-return-statement': 'error',
			'vitest/prefer-comparison-matcher': 'error',
			'vitest/prefer-expect-resolves': 'error',
			'vitest/prefer-hooks-in-order': 'error',
			'vitest/prefer-hooks-on-top': 'error',
			'vitest/prefer-to-be-falsy': 'error',
			'vitest/prefer-to-be-object': 'error',
			'vitest/prefer-to-be-truthy': 'error',
			'vitest/prefer-to-contain': 'error',
			'vitest/prefer-to-have-length': 'error',
			'vitest/valid-describe-callback': 'error',
    },
  },
];
