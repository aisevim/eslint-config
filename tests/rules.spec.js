import { execSync } from 'node:child_process'
import { join } from 'node:path'

import { describe, it } from 'vitest'

const dirname = process.cwd()

describe('Rules', () => {
  it('applied on default files, not vue, ts', ({ expect }) => {
    const target = join(dirname, 'fixtures/default')
    const config = join(dirname, 'fixtures/default/eslint.config.js')
    let output

    try {
      execSync(`eslint ${ target } -c ${ config }`, { encoding: 'utf-8' })
    } catch (error) {
      output = error.stdout
    }

    expect(output).toMatchSnapshot()
  })

  it('applied on vue, ts files', ({ expect }) => {
    const target = join(dirname, 'fixtures/vue-ts')
    const config = join(dirname, 'fixtures/vue-ts/eslint.config.js')
    let output

    try {
      execSync(`eslint ${ target } -c ${ config }`, { encoding: 'utf-8' })
    } catch (error) {
      output = error.stdout
    }

    expect(output).toMatchSnapshot()
  })
})
