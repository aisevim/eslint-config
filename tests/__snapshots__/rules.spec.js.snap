// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Rules > applied on default files, not vue, ts 1`] = `
"
/home/anon/dev/eslint-config-symmetrys/fixtures/default/exampl.yaml
  3:1  error  Empty mapping keys are forbidden  yaml/no-empty-key

/home/anon/dev/eslint-config-symmetrys/fixtures/default/example.json
  3:3  error  Duplicate key 'a'  jsonc/no-dupe-keys

/home/anon/dev/eslint-config-symmetrys/fixtures/default/index.js
   3:1   error    Module imports itself                                              import/no-self-import
   3:8   error    'file' is defined but never used                                   no-unused-vars
   7:1   warning  Expected @param names to be \\"foo\\". Got \\"fooo\\"                      jsdoc/check-param-names
  14:21  error    Use path.join() or path.resolve() instead of string concatenation  node/no-path-concat
  17:1   error    Promise.resolve() requires 0 or 1 arguments, but received 2        promise/valid-params
  21:5   error    Prefer \`.textContent\` over \`.innerText\`                            unicorn/prefer-dom-node-text-content

✖ 8 problems (7 errors, 1 warning)

"
`;

exports[`Rules > applied on vue, ts files 1`] = `
"
/home/anon/dev/eslint-config-symmetrys/fixtures/vue-ts/App.vue
  3:3   warning  \`<template>\` require directive                                        vue/no-lone-template
  3:13  error    '<template>' cannot be keyed. Place the key on real elements instead  vue/no-template-key

/home/anon/dev/eslint-config-symmetrys/fixtures/vue-ts/index.ts
   3:1  error    Module imports itself                                        import/no-self-import
   3:8  error    'file' is defined but never used                             @typescript-eslint/no-unused-vars
   7:1  warning  @param \\"fooo\\" does not match an existing function parameter  jsdoc/check-param-names
  14:1  error    Promise.resolve() requires 0 or 1 arguments, but received 2  promise/valid-params
  18:5  error    Prefer \`.textContent\` over \`.innerText\`                      unicorn/prefer-dom-node-text-content

✖ 7 problems (5 errors, 2 warnings)

"
`;
